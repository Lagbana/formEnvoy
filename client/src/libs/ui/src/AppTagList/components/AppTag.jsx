/* Single tag
=============================== */

import React from 'react';
import { Text, TouchableOpacity, StyleSheet, ViewStyle } from 'react-native';
import { Colors, Typography } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppIcon } from '../../AppIcon';
import { AppTagType } from '../AppTagList.interface';



export interface AppTagProps<T> {
  tag: AppTagType<T>;
  mode?: string;
  onPress?: (tag: AppTagType<T>) => void;
  editable?: boolean;
  style?: ViewStyle;
}

export function AppTag<T = any>({
  tag,
  mode = 'added',
  onPress = () => null,
  editable = true,
  style,
}: AppTagProps<T>) {
  // styles

  const styles = StyleSheet.create({
    item: {
      backgroundColor: editable
        ? mode === 'add'
          ? Colors.white
          : Colors.purple
        : Colors.gray + '88',
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      paddingVertical: 4,
      paddingHorizontal: 14,
      borderRadius: 30,
      marginRight: 8,
      marginBottom: 6,
      maxWidth: '100%',
    },
    label: {
      ...Typography.labelDefault,
      color: mode === 'add' ? Colors.blue : Colors.white,
    },
    icon: {
      marginLeft: 6,
      marginRight: -2,
      marginTop: 2,
      color: mode === 'add' ? Colors.green : Colors.red,
    },
  });

  return (
    <TouchableOpacity
      style={[style, styles.item]}
      onPress={() => onPress(tag)}
    >
      <Text style={styles.label}>{tag.label}</Text>
      {editable && (
        <AppIcon
          name={mode === 'add' ? 'plus' : 'x'}
          size={20}
          style={styles.icon}
        />
      )}
    </TouchableOpacity>
  );
};
