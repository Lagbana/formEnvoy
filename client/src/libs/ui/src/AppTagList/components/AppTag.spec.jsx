import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppTag } from './AppTag'

configure({ adapter: new Adapter() })

test('<AppTag /> renders correctly', () => {
  const component = shallow(
    <AppTag tag={{ label: 'test label', value: 0 }} />
  )
  expect(toJson(component)).toMatchSnapshot()
})
