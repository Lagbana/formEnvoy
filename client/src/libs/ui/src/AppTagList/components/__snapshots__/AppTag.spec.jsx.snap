// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTag /> renders correctly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "backgroundColor": "#4749A0",
        "borderRadius": 30,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 6,
        "marginRight": 8,
        "maxWidth": "100%",
        "paddingHorizontal": 14,
        "paddingVertical": 4,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "ObjektivMk1-Bold",
        "fontSize": 18,
        "lineHeight": 19.8,
        "paddingTop": 4.5,
      }
    }
  >
    test label
  </Text>
  <AppIcon
    name="x"
    size={20}
    style={
      Object {
        "color": "#FF5441",
        "marginLeft": 6,
        "marginRight": -2,
        "marginTop": 2,
      }
    }
  />
</ForwardRef>
`;
